pkg/agent/datamodel/types.go

```diff
@@ -666,6 +666,7 @@ type AgentPoolProfile struct {
 	ProximityPlacementGroupID           string               `json:"proximityPlacementGroupID,omitempty"`
 	CustomKubeletConfig                 *CustomKubeletConfig `json:"customKubeletConfig,omitempty"`
 	CustomLinuxOSConfig                 *CustomLinuxOSConfig `json:"customLinuxOSConfig,omitempty"`
+	EnableFIPS                          *bool                `json:"enableFIPS,omitempty"`
```

Comment:
let's not rely on agent pool data model. Introduce the flag in nodebootstrapconfiguration

Reply:
changed according to your comments, and added more tests